src/runtime/Loader.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from "react";
+
+interface Props {
+  render: JSX.Element | undefined;
```

Comment:
This is not an optional parameter, it is still required. This will result in a compiler error:
`<Loader />`

This will not:
`<Loader render={undefined} />`

In order for <Loader /> to succeed, you instead want to say `render?: JSX.Element`

I don't see a need to make this an optional parameter, let's just make it required:
`render: JSX.Element`

Reply:
Also let's make the type of this `(props: LoaderProps) => React.ReactNode` where the LoaderProps has the component's logs.